### ISSUE TITLE
Make the scan process fail-safe

### ISSUE THREAD

**Issue by maurodoglio** [mozilla/spade#51]
If an error happens in either the scraping or the aggregation, the database should be rolled back to the last known good state, an error logged, and the system should keep executing.

**Comment by mihneadb** [comment]
@maurodoglio Would setting autocommit to off and manually commiting the changes to the db after the successful scan make sense?

**Comment by maurodoglio** [comment]
Yes, that would be a first step, but I would also add a few fields in the db models to keep track of the status of the process. I'd like to split the scrape->parse->aggregate flow so that it can scale better and we can re-run a single task in case of failures. This is closely related to issue #49
